src/connectedk8s/azext_connectedk8s/_precheckutils.py

```diff
@@ -0,0 +1,219 @@
+        preonboarding_inspector_container_log = executing_preonboarding_inspector_job(corev1_api_instance, batchv1_api_instance, absolute_path, helm_client_location, kubectl_client_location, release_namespace, kube_config, kube_context, http_proxy, https_proxy, no_proxy, proxy_cert)
+        # If preonboarding_inspector_container_log is not empty then only we will check for the results
+        if(preonboarding_inspector_container_log is not None and preonboarding_inspector_container_log != ""):
+            preonboarding_inspector_container_log_list = preonboarding_inspector_container_log.split("\n")
+            preonboarding_inspector_container_log_list.pop(-1)
+            dns_check_log = ""
+            counter_container_logs = 1
+            # For retrieving only preonboarding inspector logs from the inspector output
+            for outputs in preonboarding_inspector_container_log_list:
+                if consts.Outbound_Connectivity_Check_Result_String in outputs:
+                    counter_container_logs = 1
+                elif consts.DNS_Check_Result_String in outputs:
+                    dns_check_log += outputs
+                    counter_container_logs = 0
+                elif counter_container_logs == 0:
+                    dns_check_log += "  " + outputs
+            dns_check = azext_utils.check_cluster_DNS(dns_check_log, True)
+            outbound_connectivity_check = azext_utils.check_cluster_outbound_connectivity(preonboarding_inspector_container_log_list[-1], True)
+        else:
+            return consts.Diagnostic_Check_Incomplete
+
+        # If both the check passed then we will return pre onboarding inspector checks Passed
+        if(dns_check == consts.Diagnostic_Check_Passed and outbound_connectivity_check == consts.Diagnostic_Check_Passed):
+            return consts.Diagnostic_Check_Passed
+        # If any of the check remain Incomplete than we will return Incomplete
+        elif(dns_check == consts.Diagnostic_Check_Incomplete or outbound_connectivity_check == consts.Diagnostic_Check_Incomplete):
+            return consts.Diagnostic_Check_Incomplete
+        else:
+            return consts.Diagnostic_Check_Failed
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to perform pre onboarding inspector container on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Pre_Onboarding_Inspector_Check_Failed_Fault_Type, summary="Error occured while performing the pre onboarding inspector container")
+
+    return consts.Diagnostic_Check_Incomplete
+
+
+def executing_preonboarding_inspector_job(corev1_api_instance, batchv1_api_instance, absolute_path, helm_client_location, kubectl_client_location, release_namespace, kube_config, kube_context, http_proxy, https_proxy, no_proxy, proxy_cert):
```

Comment:
we don't need release_namespace here since release is not even installed at this point